internal/serve/httphandler/receiver_registration.go

```diff
@@ -52,7 +52,7 @@ func (h ReceiverRegistrationHandler) ServeHTTP(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	rw, err := h.ReceiverWalletModel.GetByStellarAccountAndMemo(ctx, sep24Claims.SEP10StellarAccount(), sep24Claims.SEP10StellarMemo())
+	rw, err := h.ReceiverWalletModel.GetByStellarAccountAndMemo(ctx, sep24Claims.SEP10StellarAccount(), sep24Claims.SEP10StellarMemo(), sep24Claims.ClientDomain())
```

Comment:
I was looking into `sep24Claims.Valid()` to see if `ClientDomain()` is required there and it seems like it's not. 

The current test only checks that it's valid when not empty. 
```go
if c.ClientDomain() != "" {
  err = utils.ValidateDNS(c.ClientDomain())
  if err != nil {
	  return fmt.Errorf("client_domain is invalid: %w", err)
  }
}
```
	
This means that it's possible for ClientDomain to be empty. 

Can you please check when ClientDomain is empty?
	
And if it's acceptable that client domain is empty, maybe we should make it an optional  querying field in `GetByStellarAccountAndMemo`.